### PR TITLE
docs - add examples for GDD and add locking information for SELECT...…

### PR DESCRIPTION
**Pull request by mkiyama**
…FOR <lock_strength>

dml.xml - Added examples of GDD supported UPDATE and DELETE commands.
select.xml - added table locking information for SELECT...FOR queries.

Will be backported to 6X_STABLE

### CODE REVIEW

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml [hunk 1]
```diff
@@ -841,14 +841,19 @@ OFFSET <varname>start</varname></codeblock><p>where
...
-KEY SHARE</codeblock></p><note> By default,
-          Greenplum Database acquires an <codeph>EXCLUSIVE</codeph> lock on tables for
-            <codeph>DELETE</codeph> and <codeph>UPDATE</codeph> operations on heap tables. When the
-          Global Deadlock Detector is enabled the lock mode for <codeph>DELETE</codeph> and
+            <varname>lock_strength</varname> can be one of<ul id="ol_m3x_p54_n3b">
+            <li><codeph>FOR UPDATE</codeph> - Locks the table with an <codeph>EXCLUSIVE</codeph>
+              lock.</li>
+            <li><codeph>FOR NO KEY UPDATE</codeph> - Locks the table with an
+                <codeph>EXCLUSIVE</codeph> lock.</li>
+            <li><codeph>FOR SHARE</codeph> - Locks the table with an <codeph>ROW SHARE</codeph>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: an -> a

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml [hunk 2]
```diff
@@ -841,14 +841,19 @@ OFFSET <varname>start</varname></codeblock><p>where
...
+              lock.</li>
+            <li><codeph>FOR KEY SHARE</codeph> - Locks the table with an <codeph>ROW SHARE</codeph>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: an -> a

File: gpdb-doc/dita/admin_guide/dml.xml [hunk 3]
```diff
@@ -415,6 +415,103 @@ holder_sessionid | 8
...
+      <section id="gdd_example">
+        <title>Global Deadlock Detector UPDATE and DELETE Compatibility</title>
+        <p dir="ltr">The Global Deadlock Detector can manage concurrent updates for these types of
+            <codeph>UPDATE</codeph> and <codeph>DELETE</codeph> commands on heap tables: </p>
+        <ul id="ul_jrm_1r4_n3b">
+          <li>Simple <codeph>UPDATE</codeph> of a single table. Update a non-distribution key with
+            the Postgres Planner. The command does not contain a <codeph>FROM</codeph> clause, or a
+            sub-query in the <codeph>WHERE</codeph> clause.
+            <codeblock dir="ltr">UPDATE t SET c2 = c2 + 1 WHERE c1 > 10;</codeblock></li>
+          <li>Simple <codeph>DELETE</codeph> of single table. The command does not contain a
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/dml.xml]: of single -> of a single

File: gpdb-doc/dita/admin_guide/dml.xml [hunk 4]
```diff
@@ -415,6 +415,103 @@ holder_sessionid | 8
...
+              or references a distribution key.
+              <codeblock dir="ltr">UPDATE t SET b = b + 1 WHERE c = 10; -- c is a distribution key</codeblock></p></li>
+          <li>Complex <codeph>UPDATE</codeph>. The <codeph>UPDATE</codeph> command includes multiple
+            table
+              joins.<codeblock dir="ltr">UPDATE t1 SET c = t1.c+1 FROM t2 WHERE t1.c = t2.c;</codeblock><p
+              dir="ltr">Or the command contains a sub-query in the <codeph>WHERE</codeph>
+              clause.</p><codeblock dir="ltr">UPDATE t SET c = c + 1 WHERE c > ALL(SELECT * FROM t1);</codeblock></li>
+          <li>Complex <codeph>DELETE</codeph>. A complex <codeph>DELETE</codeph> command is similar
+            to a complex <codeph>UPDATE</codeph>, and involves multiple table joins or a
+            sub-que<codeblock>DELETE FROM t USING t1 WHERE t.c > t1.c;</codeblock></li>
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/dml.xml]: sub-que -> sub-query.

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml (hunk 1 above)
Comment (mkiyama) [reply gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: fixed

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml (hunk 2 above)
Comment (mkiyama) [reply gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: fixed

File: gpdb-doc/dita/admin_guide/dml.xml (hunk 3 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/dml.xml]: fixed

File: gpdb-doc/dita/admin_guide/dml.xml (hunk 4 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/dml.xml]: fixed

**Pull request merged**